hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineImpl.java

```diff
@@ -455,44 +459,47 @@ public void removeTrigger(String tenantId, String triggerId) {
     }
 
     private void removeTrigger(Trigger trigger) {
+        final String tenantId = trigger.getTenantId();
+        final String triggerId = trigger.getId();
+
+        // If necessary, clean up working memory
```

Comment:
Ok, then I see a potential problem
In ConditionMatch.drl#AlertOnSatisfiedDampening 

```
        if ($t.isAutoResolve()) {
            log.debugf("Setting Trigger to AutoResolve Mode! %s", $t);
            retract( $t );
            $t.setMode(Mode.AUTORESOLVE);
            insert( $t );
        }
```

this should be modified to something like

```
        if ($t.isAutoResolve()) {
            log.debugf("Setting Trigger to AutoResolve Mode! %s", $t);
            modify($t) {
               setMode(Mode.AUTORESOLVE)
            }
        }
```
To avoid conflicts with the new rules, as potentially, an autoresolve change could wipe out conditions and dampenings.

As we didn't have explicit `not Trigger` rules tjos was safe, but at the moment of use them, this could be impacted.